stores/amazon_monitoring.py

```diff
@@ -111,6 +98,33 @@
 amazon_config = {}
 
 
+class ItemsHandler:
+    @classmethod
+    def create_items_pool(cls, item_list):
+        cls.items = cycle(item_list)
+
+    @classmethod
+    def assign_next_item(cls):
+        return next(cls.items)
+
+class BadProxyCollector:
+    @classmethod
+    def __init__(cls):
+        if os.path.exists(BAD_PROXIES_PATH):
+            with open(BAD_PROXIES_PATH) as f:
+                cls.collection = json.load(f)
+        else:
+            cls.collection = {}
+
+    @classmethod
+    def collect(cls, status, connector):
+        if status == 503:
+            proxy = str(connector.proxy_url)
+            cls.collection.update({proxy : "null"})
```

Comment:
What is the structure of this collection? Is it just an object with key = `proxy_url` and value always equal to `null`? Seems like it would be better to either just use an array of urls, or have a meaningful value, like the number of 503 requests, or the time the last failure using this proxy happened.

Reply:
in fact, a value of `"null"` seems worse than a value of `null`, since `null` is null/false, but `"null"` is a four-character string.

Reply:
yeah I didn't know json automatically formats booleans and none so I threw it in as a placeholder. I've updated it to be more clear and useful.